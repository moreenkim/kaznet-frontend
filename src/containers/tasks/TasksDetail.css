.statistics-data {
  width: 977px;
}
.kaznet-title-td {
  font-weight: bold;
}

.kaznet-stats {
  background-color: #f6f6f6;
  text-align: center;
}

.kaznet-sub {
  padding-bottom: 13px;
  padding-top: 12px;
  padding-right: 0px !important;
}

.withspace {
  margin-left: 10px !important;
}

.kaznet-stats-count {
  font-weight: bold;
  font-size: 1.3em;
  margin: 0px;
}

.kaznet-stats-col {
  padding-left: 28px !important;
  padding-right: 0px !important;
  max-height: 111px !important;
}

.kaznet-no-gutter {
  padding-right: 0px !important;
}

.left-fix {
  padding-left: 0px !important;
}

.stats {
  text-align: center;
}

.stats-data {
  font-weight: bold;
  font-size: 1.5em;
  padding-top: 28px;
  line-height: 36px;
  margin: 0px;
}

.stats-header {
  margin: 0px;
  padding-bottom: 22px;
}

.details {
  text-align: left;
  margin-top: 2em;
}

.kaznet-title-h1 {
  font-size: 1.8em;
  opacity: 0.5;
}

.icon-link {
  color: #711c27;
}

.link {
  margin-left: 0.4em;
  font-size: 0.95em;
  font-weight: bold;
}

.link:hover {
  text-decoration: none;
}
