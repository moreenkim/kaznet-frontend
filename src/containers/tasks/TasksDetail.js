import React, { Component } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import Moment from "react-moment";
import _ from "lodash";
import "./TasksDetail.css";
import { rrulestr } from "rrule";
import FontAwesomeIcon from "@fortawesome/react-fontawesome";

import * as taskSelectors from "../../store/tasks/reducer";
import * as taskActions from "../../store/tasks/actions";
import * as globalActions from "../../store/global/actions";

import DetailView from "../../components/DetailView";
import StatisticsSection from "./StatisticsSection";
import NestedElementMap from "../NestedElementMap";

export class TasksDetail extends Component {
  constructor(props) {
    super(props);
    this.title = "Undefined";
  }

  componentDidMount() {
    this.props.fetchTask(this.props.match.params.id);
    this.props.changePageTitle(`Tasks > Undefined`);
  }

  componentDidUpdate() {
    this.props.changePageTitle(`Tasks > ${this.title}`);
  }

  render() {
    const task = _.find(this.props.rowById, data => {
      return data.id === this.props.match.params.id;
    });
    if (!task) return this.renderLoading();
    this.title = task.attributes.name;
    return (
      <div className="TasksList">
        <StatisticsSection
          accepted={task.attributes.approved_submissions_count}
          reward={task.attributes.total_bounty_payout}
          review={task.attributes.pending_submissions_count}
          rejected={task.attributes.rejected_submissions_count}
          totalSubmissions={task.attributes.submission_count}
        />
        <DetailView
          renderMainDetails={this.renderMainDetails(task)}
          renderAdditionalDetails={this.renderAdditionalDetails(task)}
        />
      </div>
    );
  }

  renderLoading() {
    return <p>Loading...</p>;
  }

  renderMainDetails(data) {
    const headerItems = {
      Description: data.attributes.description,
      Name: data.attributes.name,
      "Unit Reward Amount": data.attributes.current_bounty_amount,
      Form:
        data.attributes.xform_title !== ""
          ? [
              data.attributes.xform_title,
              <a href="/" key="form_link" className="link withspace">
                <FontAwesomeIcon
                  icon="external-link-alt"
                  className="fa-xs icon-link"
                  key="form_link_icon"
                />{" "}
                VIEW IN ONA
              </a>
            ]
          : "Not selected"
    };

    return <NestedElementMap detailitems={headerItems} HTMLTag="td" />;
  }

  renderAdditionalDetails(data) {
    const timingRule =
      data.attributes.timing_rule != null
        ? rrulestr(data.attributes.timing_rule).toText()
        : "";

    const headerItems = {
      "Active dates": [
        <Moment format="DD-MM-YYYY" key="start_date">
          {data.attributes.start}
        </Moment>,
        " to ",
        <Moment format="DD-MM-YYYY" key="end_date">
          {data.attributes.end}
        </Moment>
      ],
      Location: "Location Here",
      Recurring: timingRule,
      "Submission Limit": data.attributes.user_submission_target,
      "Minimum Contributor Level": data.attributes.required_expertise_display
    };

    return <NestedElementMap detailitems={headerItems} HTMLTag="td" />;
  }
}

function mapStateToProps(state) {
  return {
    rowById: taskSelectors.getTasksById(state)
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      fetchTask: taskActions.fetchTask,
      changePageTitle: globalActions.changePageTitle
    },
    dispatch
  );
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TasksDetail);
