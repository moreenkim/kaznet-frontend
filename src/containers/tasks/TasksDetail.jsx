import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Moment from 'react-moment';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import PropTypes from 'prop-types';
import '../LoadListAnimation.css';
import * as taskSelectors from '../../store/tasks/reducer';
import * as userActions from '../../store/users/actions';
import * as taskActions from '../../store/tasks/actions';
import * as globalActions from '../../store/global/actions';
import * as errorHandlerSelectors from '../../store/errorHandler/reducer';
import * as constants from '../../constants';
import ElementMap from '../ElementMap';
import DetailView from '../../components/DetailView';
import TaskDetailTitle from '../../components/tasks/TaskDetailTitle';
import StatisticsSection from './StatisticsSection';
import NestedElementMap from '../NestedElementMap';
import './TasksDetail.scss';

export class TasksDetail extends Component {
  constructor(props) {
    super(props);
    this.state = {
      modal: false
    };
    this.onFormSubmit = this.onFormSubmit.bind(this);
    this.toggle = this.toggle.bind(this);
  }

  componentDidMount() {
    const { fetchTask, noTitle, match } = this.props;
    fetchTask(match.params.id);
    noTitle();
  }

  onFormSubmit(start, end, status) {
    const { match, taskById, exportSubmissions } = this.props;

    const filterObject = {
      task: match.params.id,
      status,
      format: 'csv'
    };
    filterObject[constants.FILTER_TIME_START] = start;
    filterObject[constants.FILTER_TIME_END] = end;
    exportSubmissions(filterObject, taskById.attributes.name);
  }

  toggle() {
    const { modal } = this.state;

    this.setState({
      modal: !modal
    });
  }

  renderLoading() {
    const { hasError, errorMessage } = this.props;

    if (!hasError) {
      return (
        <center>
          <div className="lds-ripple">
            <div />
            <div />
          </div>
        </center>
      );
    }

    return <p>{errorMessage.message}</p>;
  }

  renderMainDetails() {
    const headerItems = {
      Description: this.task.attributes.description,
      Name: this.task.attributes.name,
      'Unit Reward Amount': this.task.attributes.current_bounty_amount,
      Form: this.task.attributes.xform_title
        ? [
            this.task.attributes.xform_title,
            // eslint-disable-next-line react/jsx-indent
            <a
              href={constants.ONA_LOGIN}
              target="_blank"
              rel="noopener noreferrer"
              key="form_link"
              className="link withspace"
            >
              <FontAwesomeIcon
                icon="external-link-alt"
                className="fa-xs icon-link"
                key="form_link_icon"
              />
              &nbsp;VIEW IN ONA
            </a>
          ]
        : 'Not selected'
    };

    return <NestedElementMap detailitems={headerItems} HTMLTag="td" />;
  }

  // eslint-disable-next-line class-methods-use-this
  renderLocations(locations) {
    return (
      <ul key={1} className="list-unstyled">
        <ElementMap items={locations} HTMLTag="li" />
      </ul>
    );
  }

  renderAdditionalDetails() {
    const locations = this.task.attributes.task_locations.map(el => {
      return `${el.location_name}`;
    });

    const headerItems = {
      'Active dates': [
        <Moment format="DD-MM-YYYY" key="start_date">
          {this.task.attributes.start}
        </Moment>,
        ' to ',
        <Moment format="DD-MM-YYYY" key="end_date">
          {this.task.attributes.end}
        </Moment>
      ],
      Locations: this.renderLocations(locations),
      'Submission Limit': this.task.attributes.user_submission_target,
      'Minimum Contributor Level': this.task.attributes.required_expertise_display
    };

    return <NestedElementMap detailitems={headerItems} HTMLTag="td" />;
  }

  render() {
    let xformURL;
    let xformTableURL;

    const { taskById } = this.props;
    const { modal } = this.state;
    this.task = taskById;

    if (!this.task) return this.renderLoading();

    const formOwner = this.task.attributes.xform_owner;
    const projectID = this.task.attributes.xform_project_id;
    const onaID = this.task.attributes.xform_ona_id;

    if (formOwner && projectID && onaID) {
      xformURL = `${constants.ONA_WEBSITE}/${formOwner}/${projectID}/${onaID}`;
      xformTableURL = `${xformURL}#/table`;
    } else {
      xformURL = null;
      xformTableURL = xformURL;
    }

    return (
      <div className="TasksList">
        <TaskDetailTitle task={this.task} />
        <StatisticsSection
          accepted={this.task.attributes.approved_submissions_count}
          reward={this.task.attributes.total_bounty_payout}
          review={this.task.attributes.pending_submissions_count}
          rejected={this.task.attributes.rejected_submissions_count}
          totalSubmissions={this.task.attributes.submission_count}
          task={this.task}
          formURL={xformTableURL}
          downloadModalHandler={this.toggle}
          modalState={modal}
          onFormSubmit={this.onFormSubmit}
          taskName={this.task.attributes.name}
          xformTableURL={xformTableURL}
        />
        <DetailView
          renderMainDetails={this.renderMainDetails()}
          renderAdditionalDetails={this.renderAdditionalDetails()}
        />
      </div>
    );
  }
}

TasksDetail.propTypes = {
  exportSubmissions: PropTypes.func.isRequired,
  fetchTask: PropTypes.func.isRequired,
  noTitle: PropTypes.func.isRequired,
  match: PropTypes.shape({
    params: {}
  }).isRequired,
  taskById: PropTypes.shape({
    attributes: {}
  }).isRequired,
  hasError: PropTypes.bool,
  errorMessage: PropTypes.string
};

TasksDetail.defaultProps = {
  hasError: false,
  errorMessage: ''
};

function mapStateToProps(state, props) {
  return {
    taskById: taskSelectors.getTaskById(state, props.match.params.id),
    hasError: errorHandlerSelectors.getHasError(state),
    errorMessage: errorHandlerSelectors.getErrorMessage(state)
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      fetchTask: taskActions.fetchTask,
      changeTaskStatus: globalActions.changeDetailStatus,
      noTitle: globalActions.toggleTitleOff,
      exportSubmissions: userActions.exportSubmissions
    },
    dispatch
  );
}

export default connect(mapStateToProps, mapDispatchToProps)(TasksDetail);
