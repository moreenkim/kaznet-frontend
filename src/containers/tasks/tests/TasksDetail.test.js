// Test TasksDetail
import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import { TasksDetail } from "../TasksDetail";
import * as fixtures from "../../../store/tasks/tests/fixtures";

describe("containers/task/TasksDetail", () => {
  it("renders without crashing", () => {
    shallow(
      <TasksDetail
        match={{
          params: {
            id: "1"
          }
        }}
        fetchTask={function() {}}
        changePageTitle={function() {}}
      />
    );
  });

  it("renders detail list correctly", () => {
    const wrapper = mount(
      <TasksDetail
        match={{
          params: {
            id: "1"
          }
        }}
        fetchTask={function() {}}
        changePageTitle={function() {}}
        taskById={fixtures.taskById}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
