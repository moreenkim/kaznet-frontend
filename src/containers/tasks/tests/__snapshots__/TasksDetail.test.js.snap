// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/task/TasksDetail renders detail list correctly 1`] = `
<TasksDetail
  changePageTitle={[Function]}
  fetchTask={[Function]}
  match={
    Object {
      "params": Object {
        "id": "1",
      },
    }
  }
  rowById={
    Object {
      "1": Object {
        "attributes": Object {
          "approved_submissions_count": 1,
          "created": "2018-06-21T12:15:45.349936+03:00",
          "current_bounty_amount": "55.00 KES",
          "description": "This is pure joy.",
          "end": "2019-01-30T12:17:40+03:00",
          "estimated_time": "4 01:15:20",
          "modified": "2018-06-21T12:22:09.703447+03:00",
          "name": "Awesome Task",
          "pending_submissions_count": 0,
          "rejected_submissions_count": 0,
          "required_expertise": "1",
          "required_expertise_display": "Beginner",
          "start": "2018-06-21T12:15:45+03:00",
          "status": "a",
          "status_display": "Active",
          "submission_count": 1,
          "target_id": 1,
          "timing_rule": "RRULE:FREQ=DAILY;INTERVAL=10;COUNT=12",
          "total_bounty_payout": "55.00 KES",
          "total_submission_target": 100,
          "user_submission_target": 100,
          "xform_title": "Form 1",
        },
        "id": "1",
        "relationships": Object {
          "bounty": Object {
            "data": Object {
              "id": "2",
              "type": "Bounty",
            },
          },
          "client": Object {
            "data": Object {
              "id": "7",
              "type": "Client",
            },
          },
          "locations": Object {
            "data": Array [
              Object {
                "id": "1",
                "type": "Location",
              },
            ],
            "meta": Object {
              "count": 1,
            },
          },
          "parent": Object {
            "data": null,
          },
          "segment_rules": Object {
            "data": Array [],
            "meta": Object {
              "count": 0,
            },
          },
          "target_content_type": Object {
            "data": Object {
              "id": "16",
              "type": "ContentType",
            },
          },
        },
        "type": "Task",
      },
      "2": Object {
        "attributes": Object {
          "approved_submissions_count": 2,
          "created": "2018-06-21T12:15:45.363700+03:00",
          "current_bounty_amount": "150000.00 KES",
          "description": "The best and most delicious task ever.",
          "end": null,
          "estimated_time": "00:00:19",
          "modified": "2018-06-21T12:25:06.164454+03:00",
          "name": "Kill Bill",
          "pending_submissions_count": 0,
          "rejected_submissions_count": 0,
          "required_expertise": "3",
          "required_expertise_display": "Advanced",
          "start": "2018-08-17T12:15:45+03:00",
          "status": "s",
          "status_display": "Scheduled",
          "submission_count": 2,
          "target_id": 2,
          "timing_rule": "RRULE:FREQ=DAILY;INTERVAL=1;COUNT=500",
          "total_bounty_payout": "300000.00 KES",
          "total_submission_target": 200,
          "user_submission_target": 2,
          "xform_title": "Form 2",
        },
        "id": "2",
        "relationships": Object {
          "bounty": Object {
            "data": Object {
              "id": "1",
              "type": "Bounty",
            },
          },
          "client": Object {
            "data": Object {
              "id": "2",
              "type": "Client",
            },
          },
          "locations": Object {
            "data": Array [
              Object {
                "id": "2",
                "type": "Location",
              },
            ],
            "meta": Object {
              "count": 1,
            },
          },
          "parent": Object {
            "data": null,
          },
          "segment_rules": Object {
            "data": Array [],
            "meta": Object {
              "count": 0,
            },
          },
          "target_content_type": Object {
            "data": Object {
              "id": "16",
              "type": "ContentType",
            },
          },
        },
        "type": "Task",
      },
    }
  }
>
  <div
    className="TasksList"
  >
    <StatisticsSection
      accepted={1}
      rejected={0}
      review={0}
      reward="55.00 KES"
      totalSubmissions={1}
    >
      <Col
        md={
          Object {
            "offset": 2,
            "size": 8,
          }
        }
        sm="12"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="col-sm-12 col-md-8 offset-md-2"
        >
          <Row
            tag="div"
          >
            <div
              className="row"
            >
              <Col
                className="kaznet-sub"
                sm="3"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="kaznet-sub col-sm-3"
                >
                  <p
                    className="kaznet-stats-count"
                  >
                    1
                     Submissions
                  </p>
                </div>
              </Col>
              <Col
                className="kaznet-sub"
                sm="9"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="kaznet-sub col-sm-9"
                >
                  <Button
                    color="primary"
                    tag="button"
                  >
                    <button
                      className="btn btn-primary"
                      onClick={[Function]}
                    >
                      <FontAwesomeIcon
                        border={false}
                        className=""
                        fixedWidth={false}
                        flip={null}
                        icon="clone"
                        listItem={false}
                        mask={null}
                        name=""
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        transform={null}
                      >
                        <svg
                          aria-hidden="true"
                          className="svg-inline--fa fa-clone fa-w-16 "
                          data-icon="clone"
                          data-prefix="fas"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 512 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M464 0c26.51 0 48 21.49 48 48v288c0 26.51-21.49 48-48 48H176c-26.51 0-48-21.49-48-48V48c0-26.51 21.49-48 48-48h288M176 416c-44.112 0-80-35.888-80-80V128H48c-26.51 0-48 21.49-48 48v288c0 26.51 21.49 48 48 48h288c26.51 0 48-21.49 48-48v-48H176z"
                            fill="currentColor"
                            style={Object {}}
                          />
                        </svg>
                      </FontAwesomeIcon>
                       Review all submissions
                    </button>
                  </Button>
                </div>
              </Col>
            </div>
          </Row>
          <Row
            tag="div"
          >
            <div
              className="row"
            >
              <Col
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col-sm-6"
                >
                  <Row
                    tag="div"
                  >
                    <div
                      className="row"
                    >
                      <Col
                        className="kaznet-no-gutter"
                        sm="6"
                        tag="div"
                        widths={
                          Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                        }
                      >
                        <div
                          className="kaznet-no-gutter col-sm-6"
                        >
                          <div
                            className="kaznet-stats"
                          >
                            <p
                              className="stats-data"
                            >
                              1
                            </p>
                            <p
                              className="stats-header"
                            >
                              Accepted
                            </p>
                          </div>
                        </div>
                      </Col>
                      <Col
                        className="kaznet-no-gutter left-fix"
                        sm="6"
                        tag="div"
                        widths={
                          Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                        }
                      >
                        <div
                          className="kaznet-no-gutter left-fix col-sm-6"
                        >
                          <div
                            className="kaznet-stats"
                          >
                            <p
                              className="stats-data"
                            >
                              55.00 KES
                            </p>
                            <p
                              className="stats-header"
                            >
                              Total Reward
                            </p>
                          </div>
                        </div>
                      </Col>
                    </div>
                  </Row>
                </div>
              </Col>
              <Col
                className="overflow-fix"
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="overflow-fix col-sm-6"
                >
                  <Row
                    tag="div"
                  >
                    <div
                      className="row"
                    >
                      <Col
                        className="kaznet-stats-col"
                        sm="6"
                        tag="div"
                        widths={
                          Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                        }
                      >
                        <div
                          className="kaznet-stats-col col-sm-6"
                        >
                          <div
                            className="kaznet-stats"
                          >
                            <h1
                              className="stats-data"
                            >
                              0
                            </h1>
                            <p
                              className="stats-header"
                            >
                              Needs Review
                            </p>
                          </div>
                        </div>
                      </Col>
                      <Col
                        className="kaznet-stats-col"
                        sm="6"
                        tag="div"
                        widths={
                          Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                        }
                      >
                        <div
                          className="kaznet-stats-col col-sm-6"
                        >
                          <div
                            className="kaznet-stats overflow-fix"
                          >
                            <h1
                              className="stats-data"
                            >
                              0
                            </h1>
                            <p
                              className="stats-header"
                            >
                              Rejected
                            </p>
                          </div>
                        </div>
                      </Col>
                    </div>
                  </Row>
                </div>
              </Col>
            </div>
          </Row>
          <hr />
        </div>
      </Col>
    </StatisticsSection>
    <DetailView
      renderAdditionalDetails={
        <NestedElementMap
          HTMLTag="td"
          detailitems={
            Object {
              "Active dates": Array [
                <t
                  ago={false}
                  calendar={false}
                  decimal={false}
                  element="time"
                  filter={[Function]}
                  format="DD-MM-YYYY"
                  fromNow={false}
                  interval={60000}
                  onChange={[Function]}
                  toNow={false}
                  unit={null}
                  unix={false}
                  utc={false}
            >
                  2018-06-21T12:15:45+03:00
            </t>,
                " to ",
                <t
                  ago={false}
                  calendar={false}
                  decimal={false}
                  element="time"
                  filter={[Function]}
                  format="DD-MM-YYYY"
                  fromNow={false}
                  interval={60000}
                  onChange={[Function]}
                  toNow={false}
                  unit={null}
                  unix={false}
                  utc={false}
            >
                  2019-01-30T12:17:40+03:00
            </t>,
              ],
              "Location": "Location Here",
              "Minimum Contributor Level": "Beginner",
              "Recurring": "every 10 days for 12 times",
              "Submission Limit": 100,
            }
          }
        />
      }
      renderMainDetails={
        <NestedElementMap
          HTMLTag="td"
          detailitems={
            Object {
              "Description": "This is pure joy.",
              "Form": "Form 1",
              "Name": "Awesome Task",
              "Unit Reward Amount": "55.00 KES",
            }
          }
        />
      }
    >
      <Col
        md={
          Object {
            "offset": 2,
            "size": 8,
          }
        }
        sm="15"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="col-sm-15 col-md-8 offset-md-2"
        >
          <Table
            borderless={true}
            className="kaznet-table details"
            responsiveTag="div"
            tag="table"
          >
            <table
              className="kaznet-table details table table-borderless"
            >
              <tbody>
                <NestedElementMap
                  HTMLTag="td"
                  detailitems={
                    Object {
                      "Description": "This is pure joy.",
                      "Form": "Form 1",
                      "Name": "Awesome Task",
                      "Unit Reward Amount": "55.00 KES",
                    }
                  }
                >
                  <tr
                    key="Description"
                  >
                    <ElementMap
                      HTMLTag="td"
                      className="kaznet-title-td"
                      items={
                        Array [
                          "Description",
                        ]
                      }
                      key="1"
                    >
                      <td
                        className="kaznet-title-td"
                        key="0"
                      >
                        Description
                      </td>
                    </ElementMap>
                    <ElementMap
                      HTMLTag="td"
                      items={
                        Array [
                          "This is pure joy.",
                        ]
                      }
                      key="20"
                    >
                      <td
                        key="0"
                      >
                        This is pure joy.
                      </td>
                    </ElementMap>
                  </tr>
                  <tr
                    key="Name"
                  >
                    <ElementMap
                      HTMLTag="td"
                      className="kaznet-title-td"
                      items={
                        Array [
                          "Name",
                        ]
                      }
                      key="2"
                    >
                      <td
                        className="kaznet-title-td"
                        key="0"
                      >
                        Name
                      </td>
                    </ElementMap>
                    <ElementMap
                      HTMLTag="td"
                      items={
                        Array [
                          "Awesome Task",
                        ]
                      }
                      key="40"
                    >
                      <td
                        key="0"
                      >
                        Awesome Task
                      </td>
                    </ElementMap>
                  </tr>
                  <tr
                    key="Unit Reward Amount"
                  >
                    <ElementMap
                      HTMLTag="td"
                      className="kaznet-title-td"
                      items={
                        Array [
                          "Unit Reward Amount",
                        ]
                      }
                      key="3"
                    >
                      <td
                        className="kaznet-title-td"
                        key="0"
                      >
                        Unit Reward Amount
                      </td>
                    </ElementMap>
                    <ElementMap
                      HTMLTag="td"
                      items={
                        Array [
                          "55.00 KES",
                        ]
                      }
                      key="60"
                    >
                      <td
                        key="0"
                      >
                        55.00 KES
                      </td>
                    </ElementMap>
                  </tr>
                  <tr
                    key="Form"
                  >
                    <ElementMap
                      HTMLTag="td"
                      className="kaznet-title-td"
                      items={
                        Array [
                          "Form",
                        ]
                      }
                      key="4"
                    >
                      <td
                        className="kaznet-title-td"
                        key="0"
                      >
                        Form
                      </td>
                    </ElementMap>
                    <ElementMap
                      HTMLTag="td"
                      items={
                        Array [
                          "Form 1",
                        ]
                      }
                      key="80"
                    >
                      <td
                        key="0"
                      >
                        Form 1
                      </td>
                    </ElementMap>
                  </tr>
                </NestedElementMap>
                <tr>
                  <td
                    className="kaznet-title-h1"
                  >
                    Details
                  </td>
                </tr>
                <NestedElementMap
                  HTMLTag="td"
                  detailitems={
                    Object {
                      "Active dates": Array [
                        <t
                          ago={false}
                          calendar={false}
                          decimal={false}
                          element="time"
                          filter={[Function]}
                          format="DD-MM-YYYY"
                          fromNow={false}
                          interval={60000}
                          onChange={[Function]}
                          toNow={false}
                          unit={null}
                          unix={false}
                          utc={false}
                    >
                          2018-06-21T12:15:45+03:00
                    </t>,
                        " to ",
                        <t
                          ago={false}
                          calendar={false}
                          decimal={false}
                          element="time"
                          filter={[Function]}
                          format="DD-MM-YYYY"
                          fromNow={false}
                          interval={60000}
                          onChange={[Function]}
                          toNow={false}
                          unit={null}
                          unix={false}
                          utc={false}
                    >
                          2019-01-30T12:17:40+03:00
                    </t>,
                      ],
                      "Location": "Location Here",
                      "Minimum Contributor Level": "Beginner",
                      "Recurring": "every 10 days for 12 times",
                      "Submission Limit": 100,
                    }
                  }
                >
                  <tr
                    key="Active dates"
                  >
                    <ElementMap
                      HTMLTag="td"
                      className="kaznet-title-td"
                      items={
                        Array [
                          "Active dates",
                        ]
                      }
                      key="1"
                    >
                      <td
                        className="kaznet-title-td"
                        key="0"
                      >
                        Active dates
                      </td>
                    </ElementMap>
                    <ElementMap
                      HTMLTag="td"
                      items={
                        Array [
                          Array [
                            <t
                              ago={false}
                              calendar={false}
                              decimal={false}
                              element="time"
                              filter={[Function]}
                              format="DD-MM-YYYY"
                              fromNow={false}
                              interval={60000}
                              onChange={[Function]}
                              toNow={false}
                              unit={null}
                              unix={false}
                              utc={false}
                        >
                              2018-06-21T12:15:45+03:00
                        </t>,
                            " to ",
                            <t
                              ago={false}
                              calendar={false}
                              decimal={false}
                              element="time"
                              filter={[Function]}
                              format="DD-MM-YYYY"
                              fromNow={false}
                              interval={60000}
                              onChange={[Function]}
                              toNow={false}
                              unit={null}
                              unix={false}
                              utc={false}
                        >
                              2019-01-30T12:17:40+03:00
                        </t>,
                          ],
                        ]
                      }
                      key="20"
                    >
                      <td
                        key="0"
                      >
                        <t
                          ago={false}
                          calendar={false}
                          decimal={false}
                          element="time"
                          filter={[Function]}
                          format="DD-MM-YYYY"
                          fromNow={false}
                          interval={60000}
                          key="start_date"
                          onChange={[Function]}
                          toNow={false}
                          unit={null}
                          unix={false}
                          utc={false}
                        >
                          <time
                            dateTime={"2018-06-21T09:15:45.000Z"}
                          >
                            21-06-2018
                          </time>
                        </t>
                         to 
                        <t
                          ago={false}
                          calendar={false}
                          decimal={false}
                          element="time"
                          filter={[Function]}
                          format="DD-MM-YYYY"
                          fromNow={false}
                          interval={60000}
                          key="end_date"
                          onChange={[Function]}
                          toNow={false}
                          unit={null}
                          unix={false}
                          utc={false}
                        >
                          <time
                            dateTime={"2019-01-30T09:17:40.000Z"}
                          >
                            30-01-2019
                          </time>
                        </t>
                      </td>
                    </ElementMap>
                  </tr>
                  <tr
                    key="Location"
                  >
                    <ElementMap
                      HTMLTag="td"
                      className="kaznet-title-td"
                      items={
                        Array [
                          "Location",
                        ]
                      }
                      key="2"
                    >
                      <td
                        className="kaznet-title-td"
                        key="0"
                      >
                        Location
                      </td>
                    </ElementMap>
                    <ElementMap
                      HTMLTag="td"
                      items={
                        Array [
                          "Location Here",
                        ]
                      }
                      key="40"
                    >
                      <td
                        key="0"
                      >
                        Location Here
                      </td>
                    </ElementMap>
                  </tr>
                  <tr
                    key="Recurring"
                  >
                    <ElementMap
                      HTMLTag="td"
                      className="kaznet-title-td"
                      items={
                        Array [
                          "Recurring",
                        ]
                      }
                      key="3"
                    >
                      <td
                        className="kaznet-title-td"
                        key="0"
                      >
                        Recurring
                      </td>
                    </ElementMap>
                    <ElementMap
                      HTMLTag="td"
                      items={
                        Array [
                          "every 10 days for 12 times",
                        ]
                      }
                      key="60"
                    >
                      <td
                        key="0"
                      >
                        every 10 days for 12 times
                      </td>
                    </ElementMap>
                  </tr>
                  <tr
                    key="Submission Limit"
                  >
                    <ElementMap
                      HTMLTag="td"
                      className="kaznet-title-td"
                      items={
                        Array [
                          "Submission Limit",
                        ]
                      }
                      key="4"
                    >
                      <td
                        className="kaznet-title-td"
                        key="0"
                      >
                        Submission Limit
                      </td>
                    </ElementMap>
                    <ElementMap
                      HTMLTag="td"
                      items={
                        Array [
                          100,
                        ]
                      }
                      key="80"
                    >
                      <td
                        key="0"
                      >
                        100
                      </td>
                    </ElementMap>
                  </tr>
                  <tr
                    key="Minimum Contributor Level"
                  >
                    <ElementMap
                      HTMLTag="td"
                      className="kaznet-title-td"
                      items={
                        Array [
                          "Minimum Contributor Level",
                        ]
                      }
                      key="5"
                    >
                      <td
                        className="kaznet-title-td"
                        key="0"
                      >
                        Minimum Contributor Level
                      </td>
                    </ElementMap>
                    <ElementMap
                      HTMLTag="td"
                      items={
                        Array [
                          "Beginner",
                        ]
                      }
                      key="100"
                    >
                      <td
                        key="0"
                      >
                        Beginner
                      </td>
                    </ElementMap>
                  </tr>
                </NestedElementMap>
              </tbody>
            </table>
          </Table>
        </div>
      </Col>
    </DetailView>
  </div>
</TasksDetail>
`;
